File: clients/android/sample/src/androidTest/kotlin/dev/arizona/sample/TabsE2ETest.kt
package dev.arizona.sample

import androidx.compose.ui.test.junit4.createAndroidComposeRule
import androidx.compose.ui.test.onNodeWithText
import androidx.compose.ui.test.performClick
import org.junit.Rule
import org.junit.Test

/**
 * On-device conditional rendering: navigate to /native/tabs and switch tabs;
 * the content subtree swaps (push_event -> OP_UPDATE). Run with
 * ./gradlew :sample:connectedCheck (server at 10.0.2.2:4040).
 */
class TabsE2ETest {
    @get:Rule
    val rule = createAndroidComposeRule<MainActivity>()

    @Test
    fun swapsContentWhenTabSelected() {
        rule.waitForText("Tabs")
        rule.onNodeWithText("Tabs").performClick()
        rule.waitForText("Welcome home")

        rule.onNodeWithText("About").performClick()
        rule.waitForText("About Arizona")
    }
}

File: clients/ios/Sample/UITests/TabsE2ETest.swift
import XCTest

/// On-device conditional rendering: navigate to /native/tabs and switch tabs;
/// the content subtree swaps (push_event -> OP_UPDATE). The Android analogue is
/// TabsE2ETest.
final class TabsE2ETest: XCTestCase {
    private var app: XCUIApplication!

    override func setUpWithError() throws {
        continueAfterFailure = false
        app = launchSample()
    }

    func testSwapsContentWhenTabSelected() {
        tapButton(app, "Tabs")
        waitForText(app, "Welcome home")

        tapButton(app, "About")
        waitForText(app, "About Arizona")
    }
}
